File: gateway/src/main/java/kuku/advbkm/gateway/models/BookmarkModel.java
package kuku.advbkm.gateway.models;

import lombok.*;

import java.util.HashMap;


@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@ToString
public class BookmarkModel {
    private String id;
    private String creatorID;
    private String templateID;
    private String dirID;
    private String name;
    private boolean isPublic;
    private HashMap<String, Object> data;
}

File: ui/lib/models/BookmarkModel.dart
// ignore_for_file: file_names

class BookmarkModel {
  late final String id;
  late final String creatorID;
  late final String templateID;
  late final String dirID;
  late final String name;
  late final Map<String, dynamic> data;

  BookmarkModel(this.id, this.creatorID, this.templateID, this.dirID, this.name,
      this.data);
}
